sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -70,10 +70,19 @@ abstract class Optimizer(catalogManager: CatalogManager)
   def defaultBatches: Seq[Batch] = {
     val operatorOptimizationRuleSet =
       Seq(
-        // Operator push down
-        PushProjectionThroughUnion,
+        // Optimize join
         ReorderJoin,
         EliminateOuterJoin,
+        // Simplify expressions
+        BooleanSimplification,
+        SimplifyConditionals,
+        PushFoldableIntoBranches,
+        RemoveDispensableExpressions,
+        SimplifyBinaryComparison,
+        ReplaceNullWithFalseInPredicate,
+        SimplifyConditionalsInPredicate,
```

Comment:
It seems the rules that simplify exprs still exist at the bottom (e.g., `SimplifyCasts` and `SimplifyExtractValueOps`. We don't need to move them, too?